omegaconf/omegaconf.py

```diff
@@ -787,6 +790,21 @@ def _node_wrap(
             element_type=element_type,
             ref_type=ref_type,
         )
+    elif _is_union(type_):
+        element_types = get_union_types(type_)
+        if type(None) in element_types:
+            is_optional = True
+            element_types.remove(type(None))
```

Comment:
I agree this could be a problem but, I didn't get to it because, I thought no one would type something with Union less than 2 types.

Reply:
I agree it may not be worth spending time on it => maybe you could just add an explicit check somewhere (either in the `UnionNode` constructor or in `get_union_types`?) that would raise an exception if someone tries to use `Union[type(None), T]`? (saying they should use `Optional[T]` instead)